### PR TITLE
#161201270 Setup Jest for frontend testing

### PR DESCRIPTION
**Pull request by otseobande**
**What does this PR do?**

This PR sets up the application's frontend testing environment with Jest.

**Description of Task to be completed?**

- add necessary testing packages to the application
- restructure tests directory to accommodate server and client side tests
- add jest.config.js
- update travis.yml

**How should this be manually tested?**

- Open a terminal
- Run `npm run test`

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

- [#161201270](https://www.pivotaltracker.com/story/show/161201270)

**Screenshots (if appropriate)**

N/A

**Questions:**

N/A

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  collectCoverage: true,
+  collectCoverageFrom: [
+    "resources/js/**/*.{js}",
+    "!**/node_modules/**",
+    "!**/vendor/**"
+  ],
+  coverageReporters: ['text', 'html', 'lcov'],
+  coverageDirectory: './client-coverage'
+};
```
Comment (Annmary12) [inline jest.config.js]: an extra line is needed here

File: tests/client/example.test.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+describe('example test', () => {
+  it('should work', () => {
+    expect(true).toBe(true);
+  })
+})
```
Comment (Annmary12) [inline tests/client/example.test.js]: extra line is needed here too

Comment (otseobande) [reply tests/client/example.test.js]: Noted.

File: jest.config.js (hunk 1 above)
Comment (otseobande) [reply jest.config.js]: Thank you.

File: tests/client/example.test.js (hunk 2 above)
Comment (Annmary12) [reply tests/client/example.test.js]: Great work!!!

**Review by Annmary12 — APPROVED** [review]

**Pull request merged**